### PR TITLE
Muxer

### PR DESCRIPTION
**Pull request by iwahbe**
Fixes https://github.com/pulumi/pulumi-terraform-bridge/issues/830

A PR that shows consuming this API is https://github.com/pulumi/pulumi-aws/pull/2421/files#diff-34c57e622183cb0d8dd0d3f9eaa0861b3340120e9b2ad811bac7ac7be4cea4b1.

### CODE REVIEW

**Comment by t0yv0** [comment]
Could you `make lint` on this @iwahbe 

```
 Error: Error return value of `(github.com/pulumi/pulumi-terraform-bridge/x/muxer.Main).Server` is not checked (errcheck)
```

File: pkg/tfbridge/main.go
Comment (t0yv0) [inline pkg/tfbridge/main.go:83]: Mark this experimental API in a comment please.

File: pkg/tfbridge/serve.go
Comment (t0yv0) [inline pkg/tfbridge/serve.go:37]: Nit: the glue code here can go into x/muxer.

File: pkg/tfgen/generate.go
Comment (t0yv0) [inline pkg/tfgen/generate.go:859]: This new public API is reasonable. Separate PR for this kind of thing is nice for project history (nit).

File: pf/tfgen/main.go
Comment (t0yv0) [inline pf/tfgen/main.go:81]: Mark this experimental please in the comment. Something in the API/package breakdown is a bit fishy, but not worth blocking the PR on.

File: pf/tfgen/main.go
Comment (t0yv0) [inline pf/tfgen/main.go:153]: This code is mixing ProviderInfo structures?

File: pf/tfbridge/main.go
Comment (t0yv0) [inline pf/tfbridge/main.go:133]: Mark this experimental please in the comment.

File: pf/tfbridge/main.go
Comment (t0yv0) [inline pf/tfbridge/main.go:92]: Also experimental in the comment.

File: x/muxer/mapping.go
Comment (t0yv0) [inline x/muxer/mapping.go:33]: Very much worth doing especially for basic things like both providers define the same resource. I think we have to do it. Please let's link a ticket and let's do it soon, I can help perhaps.

File: x/muxer/muxer.go
Comment (t0yv0) [inline x/muxer/muxer.go:446]: Much fun here.

File: x/muxer/main.go
Comment (t0yv0) [inline x/muxer/main.go:87]: This is not a great name for the API. Since we're in experimental its' ok we'll fix it up later.

**Comment by t0yv0** [comment]
Thanks for ramping this up quickly @iwahbe ! This is becoming a critical need for us and thank you for stepping in here. Excellent progress in short amount of time.

My overall sense is that this should go in once we have confirmation that a basic end to end use can such as AWS is working, preferably with an integration test covering the extra resources coming from AWS or PF.

The API naming and package partition strike me as a bit confused, the will need follow up work to get right so its' good to have all he added APIs marked experimental.

There is a lot of complexity in DiffConfig/Configure that AFAIK we have no urgent need for, therefore I'm glossing over reviewing those bits. It's nice to think about but we can deal with that problem when we have an actual provider with diverging configurations. Which can also inform the design, perhaps this is a kind of problem that a provider author will need to configure with more control rather than relying on a automating schema mixing, perhaps schema mixing will be useful, it's hard to anticipate.

Now the basic functionality of dispatching resources and function operations needs simple and rock solid and handle exceptions etc. I very much would like the TODOs around error handling here ticketed and done shortly after this PR lands.

The mixing of MarshallableProvider info response to the Schema needs a ticket linked to the TODO as well, we need that for tf2pulumi to work with muxed providers.

File: pkg/tfbridge/serve.go
Comment (iwahbe) [reply pkg/tfbridge/serve.go:37]: Can you explain how? The glue code here calls bridge functions like `metadata.Get` and `NewProvider`. Right now the muxer does not depend on the rest of the bridge, and I would like to keep it that way.

File: pf/tfgen/main.go
Comment (iwahbe) [reply pf/tfgen/main.go:153]: Yes. This is how we recover standard "unmapped resource/datasource" error messages.

File: x/muxer/mapping.go
Comment (iwahbe) [reply x/muxer/mapping.go:33]: https://github.com/pulumi/pulumi-terraform-bridge/issues/949

**Review by t0yv0 — APPROVED** [review]

File: pf/tests/testdata/updateprogram/Pulumi.yaml
Comment (iwahbe) [inline pf/tests/testdata/updateprogram/Pulumi.yaml:28]: `pseudoRandomNumber` is a SDKv2 defined resource, while `randomInteger` is a PF defined resource. 

This gives us a basic end-to-end integration test for the muxer.

CC @t0yv0

Comment (t0yv0) [reply pf/tests/testdata/updateprogram/Pulumi.yaml:28]: Brilliant. Thank you!

**Pull request merged**
